src/components/Menu/Menu.tsx

```diff
@@ -47,27 +49,36 @@ const Menu: React.FC<MenuProps> = ({ closeMenu, open }) => {
             <img src={SilexLogo} alt="Silex Logo" width={100} height={100} />
           </Grid>
 
-          {links.map((link, i) => (
-            <Grid item key={i}>
-              <Link
-                component={RouterLink}
-                to={link.to}
-                underline="hover"
-                color={
-                  (
-                    link.exact
-                      ? location.pathname === link.to
-                      : location.pathname.startsWith(link.to)
-                  )
-                    ? "primary"
-                    : "text.disabled"
-                }
-                onClick={closeMenu}
-              >
-                {link.text}
-              </Link>
-            </Grid>
-          ))}
+          {links.map((link, i) => {
```

Comment:
We can do two things to improve the readability here:

```jsx
links.map((link, i) => {
  (condition) ? <Link> ... </Link> : null
}
```

where `condition` is 

```javascript
!(!auth.currentProjectId && link.needProjectId)
```

_(note the negation since we want to render the link if it's we don't need the project Id)_

or

use `filter` to filter out the links in the array before mapping them:

```jsx
links.filter(link => !(!auth.currentProjectId && link.needProjectId))
     .map((link, i) => <Link> ... </Link>)
```